File: API/API/services/IProjectService.vb
Imports System.Collections.Generic

Public Interface IProjectService
    Function CreateProject(req As CreateProjectRequest, currentUserId As Guid) As ProjectResponse
    Function GetProjectsForUser(currentUserId As Guid) As List(Of ProjectResponse)
    Function GetProjectDetail(projectId As Guid, currentUserId As Guid) As ProjectResponse
    Function UpdateProject(projectId As Guid, req As UpdateProjectRequest, currentUserId As Guid) As ProjectResponse
    Sub DeleteProject(projectId As Guid, currentUserId As Guid)
    
    Function GetMembers(projectId As Guid) As List(Of MemberResponse)
    Function AddMember(projectId As Guid, req As AddMemberRequest, currentUserId As Guid) As MemberResponse
    Function UpdateMemberRole(projectId As Guid, userId As Guid, req As UpdateMemberRequest, currentUserId As Guid) As MemberResponse
    Sub RemoveMember(projectId As Guid, userId As Guid, currentUserId As Guid)
End Interface

File: API_v2/Services/Interfaces/IProjectService.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using API_v2.Models.DTOs;

namespace API_v2.Services.Interfaces
{
    public interface IProjectService
    {
        Task<ProjectResponse> CreateProjectAsync(CreateProjectRequest req, Guid currentUserId);
        Task<PagedResponse<ProjectResponse>> GetProjectsForUserAsync(Guid currentUserId, int page, int pageSize);
        Task<ProjectResponse> GetProjectDetailAsync(Guid projectId, Guid currentUserId);
        Task<ProjectResponse> UpdateProjectAsync(Guid projectId, UpdateProjectRequest req, Guid currentUserId);
        Task DeleteProjectAsync(Guid projectId, Guid currentUserId);
        Task<List<MemberResponse>> GetMembersAsync(Guid projectId, Guid currentUserId);
        Task<MemberResponse> AddMemberAsync(Guid projectId, AddMemberRequest req, Guid currentUserId);
        Task<MemberResponse> UpdateMemberRoleAsync(Guid projectId, Guid userId, UpdateMemberRequest req, Guid currentUserId);
        Task RemoveMemberAsync(Guid projectId, Guid userId, Guid currentUserId);
    }
}
